app/models/course/condition.rb

```diff
@@ -5,4 +5,6 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  NAMES = ['achievement', 'assessment', 'level']
```

Comment:
You can determine the descendants of a class. Write a class method which uses `Course::Conditions.descendants`.